// node-cricapi/index.js
var request = require("request");
var apikey = null;

exports.setAPIKey = function(_apikey) {
	apikey = _apikey;
};

exports.cricketMatches = function(callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
  request("https://cricapi.com/api/cricket?apikey=" + apikey, function(err, resp, body) { callback(err, body); });
};

exports.matches = function(callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
  request("https://cricapi.com/api/matches?apikey=" + apikey, function(err, resp, body) { callback(err, body); });
};

exports.matchCalendar = function(callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
  request("https://cricapi.com/api/matchCalendar?apikey=" + apikey, function(err, resp, body) { callback(err, body); });
};

exports.playerStats = function(pid, callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
	if(isNaN(pid)) return callback("PID should be a number. Please parse it as a number before you supply it here.");
  request("https://cricapi.com/api/playerStats?apikey=" + apikey + "&pid=" + pid, function(err, resp, body) { callback(err, body); });
};

exports.playerFinder = function(name, callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
	if(!name) return callback("Must supply player name to search for.");
  request("https://cricapi.com/api/playerFinder?apikey=" + apikey + "&name=" + name, function(err, resp, body) { callback(err, body); });
};

exports.cricketScore = function(unique_id, callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
	if(isNaN(unique_id)) return callback("unique_id should be a number. Please parse it as a number before you supply it here.");
  request("https://cricapi.com/api/cricketScore?apikey=" + apikey + "&unique_id=" + unique_id, function(err, resp, body) { callback(err, body); });
};

exports.fantasySummary = function(unique_id, callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
	if(isNaN(unique_id)) return callback("unique_id should be a number. Please parse it as a number before you supply it here.");
  request("https://cricapi.com/api/fantasySummary?apikey=" + apikey + "&unique_id=" + unique_id, function(err, resp, body) { callback(err, body); });
};

exports.fantasySquad = function(unique_id, callback) {
	if(!apikey) return callback("Invalid API Key. Please call \"setAPIKey\" and set the API key.");
	if(isNaN(unique_id)) return callback("unique_id should be a number. Please parse it as a number before you supply it here.");
  request("https://cricapi.com/api/fantasySquad?apikey=" + apikey + "&unique_id=" + unique_id, function(err, resp, body) { callback(err, body); });
};
